2011-05-25  Jerome Lambourg <lambourg@adacore.com>

       * pango/pangowin32-fontmap.c (get_family_nameW): Make sure that the
       handler to the device context is properly initialized before using it to
       find font names.
       Fix K519-012
       Submitted at https://bugzilla.gnome.org/show_bug.cgi?id=651124

2010-10-10  Arnaud Charlet  <charlet@adacore.com>

	* pango/pango-context.c (itemize_state_process_run): Hide warning,
	harmless under Windows.

2009-07-20  Arnaud Charlet  <charlet@adacore.com>

	* configure: Work around ^M handling on cygwin with igncr

2005-06-17  Arnaud Charlet  <charlet@adacore.com>

	* pango/pangowin32-fontmap.c: integrate patch from
	http://bugzilla.gnome.org/show_bug.cgi?id=110521
	Fix E607-006

--- pango/pangowin32-fontmap.c.orig	2008-02-25 08:11:40.000000000 +0100
+++ pango/pangowin32-fontmap.c	2008-04-17 18:28:43.250000000 +0200
@@ -183,7 +183,21 @@ pango_win32_inner_enum_proc (LOGFONTW   
    * Asian fonts with @ prepended to their name, ignore them.
    */
   if (lfp->lfFaceName[0] != '@')
-    pango_win32_insert_font (win32fontmap, lfp, FALSE);
+    {
+      /* Bold and italic variants can always (?) be simulated,
+       * so insert them.
+       */
+      lfp->lfItalic = 0;
+      lfp->lfWeight = FW_NORMAL;
+      pango_win32_insert_font (win32fontmap, lfp, FALSE); /* normal */
+      lfp->lfItalic = 1;
+      pango_win32_insert_font (win32fontmap, lfp, FALSE); /* italic */
+      lfp->lfItalic = 0;
+      lfp->lfWeight = FW_BOLD;
+      pango_win32_insert_font (win32fontmap, lfp, FALSE); /* bold */
+      lfp->lfItalic = 1;
+      pango_win32_insert_font (win32fontmap, lfp, FALSE); /* bold+italic */
+    }
 
   return 1;
 }
--- configure.old	2009-07-20 18:26:58.578000000 +0200
+++ configure	2009-07-20 18:27:11.026000000 +0200
@@ -23213,6 +23213,7 @@ if test "$ac_cs_awk_cr" = "a${ac_cr}b"; 
 else
   ac_cs_awk_cr=$ac_cr
 fi
+ac_cs_awk_cr=$ac_cr
 
 echo 'BEGIN {' >"$tmp/subs1.awk" &&
 _ACEOF
--- pango/pango-context.c.old	2010-10-10 19:15:06.837000000 +0200
+++ pango/pango-context.c	2010-10-10 19:15:50.328000000 +0200
@@ -1445,10 +1445,11 @@ itemize_state_process_run (ItemizeState 
 
 	  if (!g_object_get_data (G_OBJECT (fontmap), script_name))
 	    {
+#if 0
 	      g_warning ("failed to choose a font, expect ugly output. engine-type='%s', script='%s'",
 			 pango_font_map_get_shape_engine_type (fontmap),
 			 script_name);
-
+#endif
 	      g_object_set_data_full (G_OBJECT (fontmap), script_name,
 				      GINT_TO_POINTER (1), NULL);
 	    }
*** pango/pangowin32-fontmap.c.old      2011-05-25 17:48:46.102539100 +0200
--- pango/pangowin32-fontmap.c  2011-05-25 17:49:20.737304700 +0200
***************
*** 953,958 ****
--- 953,961 ----
    gint i, l;
    gsize nbytes;
  
+   /* make sure _pango_win32_hdc is initialized */
+   pango_win32_get_dc ();
+ 
    /* If lfFaceName is ASCII, assume it is the common (English) name
     * for the font. Is this valid? Do some TrueType fonts have
     * different names in French, German, etc, and does the system
